Packs/RadwareCloudDDoSProtectionServices/Integrations/RadwareCloudDDoSProtectionServices/RadwareCloudDDoSProtectionServices.py

```diff
@@ -0,0 +1,237 @@
+        end_time = continue_fetch_alerts.get('end_time')
+        start_time = continue_fetch_alerts.get('start_time')
+        skip = continue_fetch_alerts.get('fetched_alerts')
+
+    demisto.debug(f'RadwareCloudDDoS: {start_time=}, {end_time=},  {skip=}')
+    response = client.get_alerts(start_time, end_time, skip, 700)
+    documents = response.get("documents")
+    demisto.debug(f'RadwareCloudDDoS: {len(documents)=}')
+    new_continue_fetch_alerts = {}
+
+    if documents:
+        latest_timestamp_alerts = documents[0].get('context', {}).get("_timestamp")
+        demisto.debug(f'RadwareCloudDDoS: {latest_timestamp_alerts=}')
+        if len(documents) == 700:
+            demisto.debug('RadwareCloudDDoS: found next page')
+            new_continue_fetch_alerts = {'end_time': end_time, 'start_time': start_time, 'fetched_alerts': len(documents)+skip}
+
+        if not continue_fetch_alerts:
+            last_run['last_fetch_alerts'] = latest_timestamp_alerts
+            demisto.debug(f'RadwareCloudDDoS: saved {latest_timestamp_alerts=}')
+
+        if new_continue_fetch_alerts:
+            last_run['nextTrigger'] = '0'
+
+        last_run['continue_fetch_alerts'] = new_continue_fetch_alerts
+        demisto.debug(f'RadwareCloudDDoS: set {new_continue_fetch_alerts=}')
+        add_fields(documents, 'operational_alerts')
+
+    return documents, last_run
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+
+    params = demisto.params()
+    account_id: str = params.get('credentials', {}).get('identifier', '')
+    api_key: str = params.get('credentials', {}).get('password', '')
+    base_url: str = params.get('url', '').rstrip('/')
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    event_types = argToList(params.get("event_types"))
+    last_run = demisto.getLastRun()
+    demisto.debug(f'RadwareCloudDDoS: {last_run=}')
+
+    command = demisto.command()
+    demisto.info(f'Command being called is {command}')
+    try:
+        client = Client(
+            account_id=account_id,
+            api_key=api_key,
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy
+        )
+        if command == 'test-module':
+            client.get_events(take=1)
+            return_results('ok')
+        elif command == 'fetch-events':
+            events = []
+            alerts = []
+            if 'Events' in event_types:
+                events, last_run = fetch_events(client, last_run=last_run)
+            if 'Alerts' in event_types:
+                alerts, last_run = fetch_alerts(client, last_run=last_run)
+
+            demisto.setLastRun(last_run)
+            send_events_to_xsiam(events+alerts, vendor=VENDOR, product=PRODUCT)
+            demisto.debug(f'Successfully sent event {[event.get("_id") for event in events]} IDs to XSIAM')
+            demisto.debug(f'Successfully sent alert {[alert.get("_id") for alert in alerts]} IDs to XSIAM')
```

Comment:
Do you think we need this for prod? I have concerns that it could be too expansive. Maybe will print just how many alerts and events were collected?

Reply:
changed